### PR TITLE
[Fixes #5975] Geoserver backup sometime fails when running in Docker

### LINKED ISSUE #5975: Geoserver backup sometime fails when running in Docker
## Expected Behavior
Call to Geoserver B/R REST API dumps all the expected catalog data

## Actual Behavior
When running in Docker container sometime it fails the first time. On second try it seems to do the job correctly.

## Steps to Reproduce the Problem

  1. Run docker-compose services
  2. Execute backup command
  3. Inspect dumped data dir, which often will be incomplete.

### PR DESCRIPTION
**Pull request by afabiani**
- Fixes #5975 : Geoserver backup sometime fails when running in Docker

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [ ] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: geonode/br/management/commands/backup.py [hunk 1]
```diff
@@ -276,80 +276,104 @@ def create_geoserver_backup(self, settings, target_folder):
...
-                                 auth=HTTPBasicAuth(user, passwd),
-                                 timeout=10)
-                if (r.status_code == 200):
-                    gs_backup = r.json()
-                    _url = gs_backup['backups']['backup'][len(gs_backup['backups']['backup']) - 1]['href']
-                    r = requests.get(_url,
+
+        _status_code = None
+        _status_text = None
+        for _r in range(retries):
```
Comment (giohappy) [inline geonode/br/management/commands/backup.py:299]: please @afabiani add a line of comment to explain the requirement for this cycle

File: geonode/br/management/commands/settings.ini [hunk 2]
```diff
@@ -3,7 +3,7 @@ pgdump = pg_dump
 pgrestore = pg_restore
 
 [geoserver]
-datadir = geoserver/data
+datadir = /geoserver_data/data/
```
Comment (giohappy) [inline geonode/br/management/commands/settings.ini:6]: This depends on the local environment so the settings.ini should be overridable, or we should be able to pass a specific settingslini to the b/r commands. Is it possible right now @bieganowski ?

Comment (bieganowski) [reply geonode/br/management/commands/settings.ini:6]: yes, sure - in the `-c` / '--config` argument one can specify the `settings.ini` file's location

Comment (giohappy) [reply geonode/br/management/commands/settings.ini:6]: ok, from what @afabiani the settings.ini with datadir = /geoserver_data/data/ should only be the default for geonode with docker. The case without GeoNode should be reverted to "geoserver/data", which in any case will probably need to be overridden locally.

Comment (giohappy) [reply geonode/br/management/commands/settings.ini:6]: so, the proposal is to rename settings.ini to settings.ini.sample and create a settings_docker.ini.sample. 
The first one will have geoserver/data.
The latter will have the datadir as proposed in this patch.

Comment (afabiani) [reply geonode/br/management/commands/settings.ini:6]: Shall we do this before merging?

Comment (giohappy) [reply geonode/br/management/commands/settings.ini:6]: @bieganowski has created a new seprate issue for this, which will override (remove) this settings.ini, so we can merge this and then merge the next PR right after

**Pull request merged**